abba/json/include/Parse.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef PARSE_HPP
+#define PARSE_HPP
+
+#include "Json.hpp"
+#include <stdlib.h>
+
+class Parse
+{
+private:
+	std::fstream	*_file;
+	size_t			_prevPos;
+	Json			*_root;
+	Json			*_current;
+	Json			json;
+
+private:
+	enum jsonType
+	{
+		CURLY_OPEN,
+		CURLY_CLOSE,
+		STRING,
+		NUMBER,
+		BOOLEAN,
+		COLON,
+		ARRAY_OPEN,
+		ARRAY_CLOSE,
+		COMMA,
+		NULL_TYPE
+	};
+
+private:
+	struct Type
```

Comment:
This name also confused my quite a bit, seeing as it is also the name of the `Json::Type` enumeration but in this case it is a struct containing what looks to be a token